src/GraphQL/Attributes/ParserAttribute.cs

```diff
@@ -0,0 +1,91 @@
+using System.Reflection;
+using GraphQL.Types;
+
+namespace GraphQL;
+
+/// <summary>
+/// Specifies a custom parser method for a field argument or a field of an input object in a GraphQL schema.
+/// </summary>
+[AttributeUsage(AttributeTargets.Parameter | AttributeTargets.Property | AttributeTargets.Field)]
+public sealed class ParserAttribute : GraphQLAttribute
+{
+    [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods)]
+    private readonly Type? _parserType;
+    private readonly string _parserMethodName;
+    private readonly bool _includePrivate;
+
+    /// <summary>
+    /// Specifies a custom parser method for a field of an input object in a GraphQL schema using the specified parser method name.
+    /// The method must exist on the declaring type of the field and be declared as static.
+    /// The method may be private or public and must have the signature 'object <paramref name="parserMethodName"/>(object value)'.
+    /// If it is public, you may wish to mark the method with <see cref="IgnoreAttribute"/> to prevent it from being exposed in the schema.
+    /// </summary>
+    [RequiresUnreferencedCode("Please ensure the specified method is not trimmed or use an alternative constructor.")]
+    public ParserAttribute(string parserMethodName)
+    {
+        _parserMethodName = parserMethodName
+            ?? throw new ArgumentNullException(nameof(parserMethodName));
+        _includePrivate = true;
+    }
+
+    /// <summary>
+    /// Specifies a custom parser method for a field of an input object in a GraphQL schema using the 'Parser' method from a specified type.
+    /// The method must exist on the specified type and must be static and public.
+    /// The method must have the signature 'object Parser(object value)'.
+    /// </summary>
+    public ParserAttribute(
+        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods)]
+        Type parserType)
+    {
+        _parserType = parserType
+            ?? throw new ArgumentNullException(nameof(parserType));
+        _parserMethodName = nameof(FieldType.Parser);
+    }
+
+    /// <summary>
+    /// Specifies a custom parser method for a field of an input object in a GraphQL schema using the specified method from a specified type.
+    /// The method must exist on the specified type and must be static and public.
+    /// The method must have the signature 'object <paramref name="parserMethodName"/>(object value)'.
+    /// </summary>
+    public ParserAttribute(
+        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods)]
+        Type parserType, string parserMethodName)
```

Comment:
@Shane32, I think, when the `parserType` defines the same type where `Parser` attribute is applied (even though it can be omitted), it should also set `_includePrivate = true`

Reply:
Sure; makes sense